src/ansi-c/c_typecheck_base.h

```diff
@@ -148,10 +148,15 @@ class c_typecheck_baset:
   // contracts
   virtual void typecheck_spec_function_pointer_obeys_contract(exprt &expr);
   virtual void typecheck_spec_assigns(exprt::operandst &targets);
-  virtual void typecheck_spec_assigns_condition(exprt &condition);
+  virtual void typecheck_conditional_targets(
+    exprt::operandst &targets,
+    const std::function<void(exprt &)> typecheck_target,
```

Comment:
It seems `typecheck_conditional_targets` is called exactly once, making the need for this configurable function unnecessary?!

Reply:
this is is mean to allow checking both assigns and frees-clauses targets (next PR)